src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_DeleteProfileScreen.tsx

```diff
@@ -34,8 +35,25 @@ export const Profile_DeleteProfileScreen = ({
 
   const [deleteError, setDeleteError] = useState<boolean>(false);
 
-  const doDeleteProfile = async () => {
-    await Alert.alert(
+  const {deleteCollectedData} = useBeaconsState();
+
+  const handleDeleteProfile = async () => {
+    try {
+      const deleteOK = await deleteProfile();
+      if (deleteOK) {
+        await deleteCollectedData();
```

Comment:
The internal functions handle if beacons is enabled or not for the organization and if is already initialized.